### PR TITLE
WebDriver/Selenium update to support Firefox 59

### PR DESCRIPTION
**Pull request by rvbernal**
Using Selenium 3.12.0 and latest web drivers

### CODE REVIEW

File: plugins/webdriver/nbactions-release-profile.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (undera) [inline plugins/webdriver/nbactions-release-profile.xml]: Looks like this file is something not needed. Can you remove it?

File: plugins/webdriver/pom.xml [hunk 2]
```diff
@@ -4,197 +4,199 @@
 
     <groupId>kg.apc</groupId>
     <artifactId>jmeter-plugins-webdriver</artifactId>
-    <version>2.3</version>
+    <version>2.4</version>
 
     <name>WebDriver/Selenium</name>
-    <description>WebDriver/Selenium</description>
```
Comment (undera) [inline plugins/webdriver/pom.xml]: Please remove all indentation changes, so only the actual XML changes will be visible.

**Comment by undera** [comment]
Please look at unit tests that are failed. Failed UT most likely means some functionality is broken. The rule of project is to have UT passed.

File: plugins/webdriver/nbactions-release-profile.xml (hunk 1 above)
Comment (rvbernal) [reply plugins/webdriver/nbactions-release-profile.xml]: Thanks, will remove that file, also, just found out that failed UTs are due to deprecation of methods and properties in Selenium 3.12.0. Will fix UTs.

File: plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/ChromeDriverConfig.java [hunk 3]
```diff
@@ -6,17 +6,17 @@
 import org.openqa.selenium.chrome.ChromeDriver;
```
Comment (undera) [inline plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/ChromeDriverConfig.java]: Changes in this file seem to be unneeded

File: plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/FirefoxDriverConfig.java [hunk 4]
```diff
@@ -63,11 +64,7 @@ private void addExtensions(FirefoxProfile profile) {
         for (int i = 0; i < rows.size(); i++) {
             ArrayList row = (ArrayList) rows.get(i).getObjectValue();
             String filename = ((JMeterProperty) row.get(0)).getStringValue();
-            try {
-                profile.addExtension(new File(filename));
-            } catch (IOException e) {
-                log.error("Failed to add extension " + filename, e);
-            }
+            profile.addExtension(new File(filename));
```
Comment (undera) [inline plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/FirefoxDriverConfig.java]: Why change this? Does it stop to throw exceptions in newer Selenium?

File: plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/FirefoxDriverConfig.java [hunk 5]
```diff
@@ -97,7 +94,7 @@ private void setPreferences(FirefoxProfile profile) {
 
     @Override
     protected FirefoxDriver createBrowser() {
-        return new FirefoxDriver(new FirefoxBinary(), createProfile(), createCapabilities());
+        return new FirefoxDriver();
```
Comment (undera) [inline plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/FirefoxDriverConfig.java]: Why change this? Will it lose some functionality?

File: plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/RemoteDesiredCapabilitiesFactory.java [hunk 6]
```diff
@@ -12,7 +12,7 @@ public static DesiredCapabilities build(RemoteCapability capability){
 	  if(RemoteCapability.CHROME.equals(capability)){
 		  ChromeOptions options = new ChromeOptions();
 	      desiredCapabilities = DesiredCapabilities.chrome();
-	      desiredCapabilities.setCapability(ChromeOptions.CAPABILITY, options);
+//	      desiredCapabilities.setCapability(ChromeOptions.CAPABILITY, options);
```
Comment (undera) [inline plugins/webdriver/src/main/java/com/googlecode/jmeter/plugins/webdriver/config/RemoteDesiredCapabilitiesFactory.java]: Why did you disable this?

**Comment by jloisel** [comment]
Would be nice to upgrade to 3.13.1 as it's available now!

**Comment by undera** [comment]
Closing in favor of #298

**Pull request closed without merging**